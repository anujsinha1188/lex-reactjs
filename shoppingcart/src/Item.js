import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
class Item extends React.Component {
    constructor({ quantity }) {
        super()
        this.state = {
            itemids: [],
            quant: quantity,
            errorMessage: "",
            name: ""
        }
    }
    handleQuantityChange(event, maxquantity, name) {
        const newQuantity = event.target.value
        if (newQuantity > parseInt(maxquantity)) {
            var message = "You cannot buy more than " + maxquantity + " " + name
            this.setState({ errorMessage: "You cannot buy more than " + maxquantity + " " + name })
        }
        this.setState({ quant: newQuantity })
        this.props.quantityChange(message)
    }

    addItem(id, price) {
        const newQuantity = this.state.quant
        this.props.callbackParent(id, newQuantity, price)
    }

    render() {
        var id = this.props.itemid
        var maxquantity = this.props.maxquantity
        return (
            <React.Fragment>
                <h1>Item Component</h1>
                <tr>
                    <td style={{ width: "30%" }}>
                        <div className="inline-block">{this.props.name}</div>
                        <div className="inline-block spacing">
                            <img width="100px" height="100px" src={this.props.img} />
                        </div>
                    </td>
                    <td style={{ width: "10%" }}>
                        <div class="align-center">{this.props.price}</div>
                    </td>
                    <td style={{ width: "10%" }}>
                        <div className="align-center"><input className="input-width" id="quantity" type="number" value={this.state.quant} onChange={(e) => this.handleQuantityChange(e, maxquantity, this.props.name)} /></div>
                    </td>
                    <td style={{ width: "10%" }}>
                        <div className="align-center">
                            <button className="btn btn-info" id="addButton" type="button" onClick={() => this.addItem(id, this.props.price)}>Add</button>
                        </div>
                    </td>
                </tr>
            </React.Fragment>
        )
    }
}
export default Item;
