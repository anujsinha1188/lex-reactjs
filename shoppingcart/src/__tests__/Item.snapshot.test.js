import Item from '../Item.js';
import renderer from 'react-test-renderer';
import React from 'react';

test('Item component renders the props correctly', () => {
  const itemDetail = {
    "itemId":"1",
    "itemName":"Chocalates",
    "ItemImg":"/Images/chocalates.png",
    "itemPrice":"100",
    "itemQuantity":"1",
    "MaxQuantity":"10"
}
const myFun1 = jest.fn()
const myFun2 = jest.fn()
  const rendered = renderer.create(
    <Item  itemid={itemDetail.itemId}
    name={itemDetail.itemName}
    img={itemDetail.ItemImg}
    price={itemDetail.itemPrice}
    quantity={itemDetail.itemQuantity}
    maxquantity={itemDetail.MaxQuantity}
    callbackParent={(id, quant, price) => myFun1(id, quant, price)}
    quantityChange={(message) => myFun2(message)} />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});