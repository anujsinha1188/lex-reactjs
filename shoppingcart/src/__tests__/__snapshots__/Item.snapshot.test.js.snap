// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component renders the props correctly 1`] = `
<tr>
  <td
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <div
      className="inline-block"
    >
      Chocalates
    </div>
    <div
      className="inline-block spacing"
    >
      <img
        height="100px"
        src="/Images/chocalates.png"
        width="100px"
      />
    </div>
  </td>
  <td
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <div
      class="align-center"
    >
      100
    </div>
  </td>
  <td
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <div
      className="align-center"
    >
      <input
        className="input-width"
        id="quantity"
        onChange={[Function]}
        type="number"
        value="1"
      />
    </div>
  </td>
  <td
    style={
      Object {
        "width": "10%",
      }
    }
  >
    <div
      className="align-center"
    >
      <button
        className="btn btn-info"
        id="addButton"
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </div>
  </td>
</tr>
`;
